### PR TITLE
Raffle docs

### LINKED ISSUE #63: Raffle contract documentation
Prepare documentation for Raffle contract.

### LINKED ISSUE #64: It's possible to buy tickets after raffle is played in Raffle contract
After the `play_raffle` method is invoked it should not be possible to buy any new tickets.

### PR DESCRIPTION
**Pull request by mariopil**
**Referenced issue**

Closes #63 and closes #64 

**Summary of changes**

Added Raffle contract documentation, fixed a bug related with buying a ticket after the raffle has been played.

### CODE REVIEW

File: contracts/raffle/src/lib.rs [hunk 1]
```diff
@@ -128,6 +128,10 @@ impl RaffleContract {
             return Err(Error::NotInitialized);
         }
 
+        if storage.get::<_, bool>(&DataKey::AlreadyPlayed).unwrap() {
+            return Err(Error::AlreadyPlayed);
+        }
+
```
Comment (eloylp) [inline contracts/raffle/src/lib.rs:134]: Nitpick:

Exactly this checking is also repeated in other methods. I consider that code a bit ugly and difficult to read. An alternative could be wrap it in a function like:

```rust
fn must_not_be_already_played() -> Result<()> {
       if storage.get::<_, bool>(&DataKey::AlreadyPlayed).unwrap() {
            return Err(Error::AlreadyPlayed);
        }
}
```
Then we could reuse that function:
```rust
fn my_contract_function_1() -> Result<(), Error>{
// ...
must_not_be_already_played()?;
//...
}
fn my_contract_function_2(){
// ...
must_not_be_already_played().unwrap();
//...
}
```

File: contracts/raffle/src/test.rs [hunk 2]
```diff
@@ -175,6 +175,35 @@ fn buy_ticket_panics_if_buyer_has_not_enough_funds() {
...
+    let contract_id = env.register_contract(None, RaffleContract);
+    let client = RaffleContractClient::new(&env, &contract_id);
+    let token_admin = Address::random(&env);
+    let test_token_client = create_token_contract(&env, &token_admin);
+
+    client.init(&client.address, &test_token_client.address, &1, &100);
+
+    let ticket_buyer_1 = Address::random(&env);
+    let ticket_buyer_2 = Address::random(&env);
+    let ticket_buyer_3 = Address::random(&env);
```
Comment (eloylp) [inline contracts/raffle/src/test.rs:193]: Maybe we can reduce this to only one buyer and reduce code in the test :)

**Review by eloylp — APPROVED** [review]
Great docs, little nitpicks. Thanks for this !

File: app/pages/contracts/raffle.md [hunk 3]
```diff
@@ -1,3 +1,70 @@
...
\ No newline at end of file
+The Raffle contract implements a simple raffle: users buy tickets and one or more of them are randomly selected as winners. Prizes are paid out from the pool filled with tokens from bought tickets, evenly distributed among all winners.
+
+## Features
+* everyone can join the raffle by buying the ticket,
+* each player can buy many tickets - the more he buys, the more chanes of win he has,
+* there is always a winner in the raffle,
+* once the raffle is finished, no more tickets can be bought.
+
+## Importing with nebula-importer
```
Comment (geofmureithi) [inline app/pages/contracts/raffle.md:11]: Needs a linebreak for valid markdown
```suggestion
## Importing with nebula-importer

```

File: app/pages/contracts/raffle.md [hunk 4]
```diff
@@ -1,3 +1,70 @@
...
+
+## Using the contract
+### Contract initialization
+To initialize the contract the `init` method needs to be called with 4 arguments:
+* `admin` - address of the admin account
+* `token` - ID of the token contract
+* `max_winners_count` - number of players that could win a raffle (1 is minimum)
+* `ticket_price` - price of the ticket
+
+```
```
Comment (geofmureithi) [inline app/pages/contracts/raffle.md:30]:
```suggestion
```bash
```

File: app/pages/contracts/raffle.md [hunk 5]
```diff
@@ -1,3 +1,70 @@
...
+    --token ${token_contract_id} \
+    --max_winners_count 1 \
+    --ticket_price 5001
+```
+Contract can only be initialized once.
+### Buying tickets
+Each user that wants to take a part in the raffle need to buy a ticket first. To do this a `buy_ticket` method need to be invoked with the following argument:
+* `by` - address of a player
+
+```
```
Comment (geofmureithi) [inline app/pages/contracts/raffle.md:47]:
```suggestion
```bash
```

File: app/pages/contracts/raffle.md [hunk 6]
```diff
@@ -1,3 +1,70 @@
...
+    -- \
+    buy_ticket \
+      --by ${player_address}
+```
+Ticket can only be bought once by a user.
+### Playing the raffle
+Only user with `admin` role can start the raffle with the `play_raffle` method that requires one argument:
+* `random_seed` - a seed used to initialize random number generator
+
+```
```
Comment (geofmureithi) [inline app/pages/contracts/raffle.md:61]:
```suggestion
```bash
```

File: app/pages/contracts/raffle.md [hunk 7]
```diff
@@ -1,3 +1,70 @@
...
+* there is always a winner in the raffle,
+* once the raffle is finished, no more tickets can be bought.
+
+## Importing with nebula-importer
+````toml
+[package.metadata.nebula.imports]
+raffle = "ghcr.io/eigerco/nebula/contracts/raffle:latest"
+````
+
+## Contract methods
```
Comment (geofmureithi) [inline app/pages/contracts/raffle.md:17]:
```suggestion
## Contract methods

```

**Review by geofmureithi — COMMENTED** [review]
Not sure those tests should in this pull.
LGTM, left some suggestions especially affecting the documentation page itself

**Comment by geofmureithi** [comment]
The markdown is still not correctly formatted.
Well I think its because its blocked by #57

**Comment by mariopil** [comment]
What is wrong with it?

**Comment by geofmureithi** [comment]
I have the code highlighting tied to the #57 PR.
Also the nebula-importer part would also need to be merged too to be valid.
Maybe I should split it.

File: app/pages/contracts/raffle.md [hunk 8]
```diff
@@ -1,3 +1,80 @@
...
+# Raffle contract
 
-Here it goes
\ No newline at end of file
+The Raffle contract implements a simple raffle: users buy tickets and one or more of them are randomly selected as winners. Prizes are paid out from the pool filled with tokens from bought tickets, evenly distributed among all winners.
+
+## Features
+
+* everyone can join the raffle by buying the ticket,
+* each player can buy many tickets - the more he buys, the more chanes of win he has,
```
Comment (geofmureithi) [inline app/pages/contracts/raffle.md:8]:
```suggestion
* each player can buy many tickets - the more he buys, the more chances of win he has,
```

File: app/pages/contracts/raffle.md [hunk 9]
```diff
@@ -1,3 +1,80 @@
...
+```bash
+soroban contract invoke \
+  --id ${contract_id} \
+  --source ${admin_address} \
+  --network ${network} \
+  -- \
+  play_raffle \
+    --random_seed 1234
+```
+After this method is called a winner(s) is randomly selected and the raffle prize is paid out. Additionaly an event with winner(s) address(es) and the value of pay out is emitted. No more tickets can be bought after the raffle has been played.
```
Comment (geofmureithi) [inline app/pages/contracts/raffle.md:80]:
```suggestion
After this method is called a winner(s) is randomly selected and the raffle prize is paid out. Additionally an event with winner(s) address(es) and the value of pay out is emitted. No more tickets can be bought after the raffle has been played.
```

**Review by geofmureithi — APPROVED** [review]
There are a few typos, but this is unblocked.

**Pull request merged**